File: Stack.py
class Node:
    def __init__(self, value=0):
        self.data = value
        self.next = None

class Stack:

    def __init__(self):
        self.head = None

    def pop(self):
        if self.head != None:
            current = self.head.data
            self.head = self.head.next
            return current
        else:
            return None

    def push(self, value):
        current = Node(value)
        if self.head == None:
            self.head = current
        else:
            current.next = self.head
            self.head = current

    def peak(self):
        if self.head != None:
            return self.head.data
        else:
            return None

File: Lab 3/Будило Е.А/scr/Stack.cpp
#include "stack.h"
Stack::Stack(int stackSize)
{
    size = stackSize;
    node = new AVL::Node * [size];
    top = -1;
}
Stack::~Stack()
{
    delete[] node;
}
void Stack::push(AVL::Node* x)
{
    if (top == size - 1)
    {
        AVL::Node** runBuff = new AVL::Node * [size];
        for (int i = 0; i < size; ++i)
            runBuff[i] = node[i];
        delete[] node;
        int sizeBuff = size;
        size *= 2;
        node = new AVL::Node * [size];
        for (int i = 0; i < sizeBuff; ++i)
            node[i] = runBuff[i];
        delete[] runBuff;
    }
    node[++top] = x;
}
AVL::Node* Stack::pop()
{
    return node[top--];
}
AVL::Node* Stack::peak()
{
    return node[top];
}
bool Stack::empty()
{
    return top == -1;
}